### ISSUE TITLE
Failing to refresh cached repo with merge errors

### ISSUE THREAD

**Issue by forketyfork** [Hacklone/private-bower#171]
private-bower sometimes fails to refresh certain cached repositories. It looks like this in the logs:
```
[bower]  1/9/15 09:46:11   Error during "git pull" in "/var/private-bower/gitRepoCache/lodash". Output was: Auto-merging lodash.core.min.js
CONFLICT (content): Merge conflict in lodash.core.min.js
Auto-merging lodash.core.js
Automatic merge failed; fix conflicts and then commit the result.

[bower]  1/9/15 09:46:11   Command failed: /bin/sh -c git pull

undefined
```
I suppose that this happens when a repository history was overwritten by its authors at some point in time, so private-bower can't automatically merge or reset the cached repo. Manually resetting the last local commit solves the problem:

```
git reset --hard HEAD~1
git pull
```

I saw this happen on several cached repos of popular libraries (lodash, jointjs, bootstrap-fileinput) for the last week, so such history rewrites might appear quite often. Possibly, to avoid this, instead of `pull`ing, private-bower should do a hard reset of the master branch, like this:
```
git fetch --all
git reset --hard origin/master
```

**Comment by forketyfork** [comment]
Oops, I've missed the pr that was supposed to fix this: #93
But I still see this issue on my system. I have the latest versions of node, private-bower and git:

```
# node --version
v0.12.7
# private-bower --version
1.1.4
# git --version
git version 2.5.1
```

**Comment by szabyg** [comment]
This should be closed. With the next release only?
